trail/trail.go

```diff
@@ -138,14 +138,18 @@ func FromGRPC(err error, args ...interface{}) error {
 	case codes.Unimplemented:
 		e = &trace.NotImplementedError{Message: message}
 	default:
-		e = errors.New(message)
+		e = err
 	}
 	if len(args) != 0 {
 		if meta, ok := args[0].(metadata.MD); ok {
 			e = DecodeDebugInfo(e, meta)
+			if _, ok := e.(trace.Error); ok {
```

Comment:
Why are we returning here if `e` is a trace error?

Reply:
Added an explanatory comment